AmplifyPlugins/API/AWSAPICategoryPlugin/Core/AppSyncList.swift

```diff
@@ -43,9 +117,20 @@ public class AppSyncList<ModelType: Model>: List<ModelType>, ModelListDecoder {
         if case let .object(jsonObject) = json,
            case .array = jsonObject["items"] {
             return true
+        } else if case let .object(jsonObject) = json,
+                  case .string = jsonObject["associatedId"],
+                  case .string = jsonObject["associatedField"],
+                  case .string(let listType) = jsonObject["listType"],
+                  listType == "appSyncList" {
+            return true
         }
 
-        return false
+        do {
+            _ = try AppSyncListPayload.init(from: decoder)
+            return true
+        } catch {
+            return false
```

Comment:
please see latest AppSyncListDecoder's shouldDecode method, will update the documentation and explain the flow better

There will be times when there exist both DataStoreListDecoder and AppSyncListDecoder registered. This means, when the developer tries to create a list or decode to a model that contains a list, the `shouldDecode` will run both both decoders registered. the normal call pattern is to return false from the AppSyncListDecoder because the developer is attempt to decode some payload from a SQL store. By checking for `AppSyncListPayload`, `ModelMetadata`, and `AppSyncListResponse`, these are the conditions when a list can be successfully created for an AppSync data source.

`AppSyncListPayload` is created by the APIPlugin's decoder logic when it detects that the response is a List type and adds extra data to the payload to preserve the request.
`ModelMetadata` is added to nested levels of List objects that contains the associated parent data.
`AppSyncListResponse` is the list response structure of the AppSync service for the list operation, containing "items" and "nextToken".